### PR TITLE
Feature Noise XKpsk3 integration topology

### PR DESCRIPTION
**Pull request by simonwicky**
# Description

This is a part of stacked PRs to integrate XKpsk3 Noise connection between nym nodes

This PR concerns Topology Access for nodes

The stack of PR is as follow :
#4360 -> #4361  (this one) -> #4373 -> #4362

# Checklist:

- [ ] added a changelog entry to `CHANGELOG.md`

### CODE REVIEW

File: common/client-core/src/client/topology_control/nym_api_provider.rs [hunk 1]
```diff
@@ -77,13 +77,23 @@ impl NymApiTopologyProvider {
             Ok(gateways) => gateways,
         };
 
+        let nodes_described = match self.validator_client.get_cached_described_nodes().await {
```
Comment (octol) [inline common/client-core/src/client/topology_control/nym_api_provider.rs:80]: `.tap_err(|err| error!("...")).ok()?`

File: gateway/Cargo.toml [hunk 2]
```diff
@@ -78,6 +78,8 @@ nym-task = { path = "../common/task" }
 nym-types = { path = "../common/types" }
 nym-validator-client = { path = "../common/client-libs/validator-client" }
 nym-ip-packet-router = { path = "../service-providers/ip-packet-router" }
+nym-client-core = { path = "../common/client-core"}
```
Comment (octol) [inline gateway/Cargo.toml]: Can we move the common stuff somewhere else so we don't have to import client-core?

File: mixnode/Cargo.toml [hunk 3]
```diff
@@ -59,6 +59,7 @@ nym-pemstore = { path = "../common/pemstore", version = "0.3.0" }
 nym-task = { path = "../common/task" }
 nym-types = { path = "../common/types" }
 nym-topology = { path = "../common/topology" }
+nym-client-core = { path = "../common/client-core/" }
```
Comment (octol) [inline mixnode/Cargo.toml]: See other comment about client-core

**Review by octol — APPROVED** [review]

File: common/client-core/src/client/topology_control/nym_api_provider.rs (hunk 1 above)
Comment (simonwicky) [reply common/client-core/src/client/topology_control/nym_api_provider.rs:80]: I copied the existing code structure for visibility

File: gateway/Cargo.toml (hunk 2 above)
Comment (simonwicky) [reply gateway/Cargo.toml]: Object of PR #4373

File: mixnode/Cargo.toml (hunk 3 above)
Comment (simonwicky) [reply mixnode/Cargo.toml]: Object of PR #4373

**Pull request merged**